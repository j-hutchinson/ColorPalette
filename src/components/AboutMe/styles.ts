import styled from 'styled-components'

export const StyledWrapper = styled.div`
    display: grid;    
    grid-template-columns: 0.3fr 0.7fr;
    gap: 48px;
    min-height: 400px;
    
    @media (max-width: 800px) {
        grid-template-columns: 1fr;
    }
`;

export const StyledBio = styled.div`
    letter-spacing: 0.25px;
    line-height: 24px;
    font-size: 18px;
`;

export const StyledLinks = styled.div`
    padding-top: 24px;
`;

export const StyledLink = styled.a`
    display: inline;
    margin: 16px 16px 16px 0;
    color: #3B70A2;
    text-decoration: none;
    font-style: italic;

    :hover {
        text-decoration: underline;
    }
`;