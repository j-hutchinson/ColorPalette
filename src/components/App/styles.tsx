import styled from 'styled-components';

export const StyledAppWrapper = styled.div`
    max-width: 1360px;
    margin: auto;
    position: relative;
`;

export const StyledContainer = styled.div`
    display: grid;
    grid-template-columns: 1fr 1fr 1fr;
    grid-template-rows: 1fr 1fr 1fr;
    gap: 24px;
    width: 100%;
    margin-top: 48px;
`;

export const StyledItem = styled.div`
    background: ${(props: { hex: string }) => props.hex};
    border-radius: 20px;
    box-shadow: 0 0 14px ${(props: { hex: string }) => props.hex};
    display: flex;
    flex-direction: column;
    height: 200px;

    :hover {
        box-shadow: 0 0 30px ${(props: { hex: string }) => props.hex};
    }
`;

export const StyledText = styled.div`
    margin: auto;
    text-align: center;
`;

export const StyledHeader = styled.h2`
    font-size: 20px;
    margin: 8px auto;
`;

export const StyledLabel = styled.p`
    color: black;
    margin: auto;
    font-size: 16px;
    text-transform: uppercase;
`;