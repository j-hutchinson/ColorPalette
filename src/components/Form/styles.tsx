import styled from 'styled-components';

export const StyledForm = styled.form`
    display: flex;
    width: 100%;
    justify-content: center;
    border-bottom: 1px solid #DCDCDC;
    margin: 48px auto;
    padding-bottom: 40px;
`;

export const StyledInput = styled.input`
    border-radius: 8px 0px 0px 8px;
    border: 1px solid #E7E7E7;
    font-family: 'Roboto' sans-serif;
    font-size: 20px;
    height: 40px;
    padding: 4px 24px;
    width: 200px;
    margin: auto;
`;

export const StyledLabel = styled.p`
    margin: 0;
`;

export const StyledSubmit = styled.input`
    padding: 12px 24px;
    font-family: 'Roboto' sans-serif;
    background: #DCDCDC;
    border: 1px solid #E7E7E7;
    border-radius: 0px 8px 8px 0px;
    font-size: 20px;
`;

export const StyledIcon = styled.span`
    color: #3D3D3D;
    margin: auto 0 auto 24px;
`;