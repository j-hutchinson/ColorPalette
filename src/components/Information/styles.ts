import styled from 'styled-components'

export const StyledContainer = styled.div`
    display: grid;
    grid-template-columns: 1fr 1fr;
    gap: 24px;

    @media screen and (max-width: 1024px) {
        grid-template-columns: 1fr;
        grid-template-rows: 1fr 1fr;
    }
`;

export const StyledBox1 = styled.div`
    background: radial-gradient(118.99% 670.46% at -7.06% -9.5%,#FFFFFF 0%,#F1F1F1 100%);
    box-shadow: 0px 0px 10px 5px rgb(0 0 0 / 10%);
    border-radius: 8px;
`;

export const StyledBox2 = styled.div`
    display: grid;
    grid-template-rows: 1fr 1fr;
    gap: 24px;
`;

export const StyledInner = styled.div`
    background: radial-gradient(118.99% 670.46% at -7.06% -9.5%,#FFFFFF 0%,#F1F1F1 100%);
    border-radius: 8px;
    box-shadow: 0px 0px 10px 5px rgb(0 0 0 / 10%);
    font-style: italic;
    font-weight: 500;
    padding: 24px;
`;

export const StyledList = styled.ul`
    display: flex;
    gap: 24px;
    font-style: normal;
    list-style: none;
    padding: 0;
`;

export const StyledJobList = styled.ul`
    display: block;
    list-style: none;
    padding: 0;
`;

export const StyledListItem = styled.li`
    font-style: normal;
    :before {
        content: "ðŸ‘‰";
        margin-right: 8px;
    }
`;

export const StyledJobItem = styled.li`
    width: 100%;
    display: block;    
    
    :first-child {
        border-bottom: 1px solid #E6E7EC;
    }    
`;