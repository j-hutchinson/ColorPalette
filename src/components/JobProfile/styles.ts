import styled from 'styled-components'

export const StyledContainer = styled.div`
    margin: 32px 0;
`;

export const StyledCompanyName = styled.h3`
    margin: 0px;
    font-size: 22px;
    display: inline;
`;
export const StyledTitle = styled.h4`
    margin: 8px 0px 16px;
    color: #878787;
    font-size: 18px;
    font-style: normal;
`;
export const StyledTop = styled.div`
    display: flex;
    justify-content: space-between;
    margin-top: 20px;
    font-style: normal;
`;

export const StyledDate = styled.p`
    display: inline;
    font-size: 18px;
    margin: 4px 0px 0px;
`;

export const StyledDescription = styled.p`
    display: block;
    font-size: 18px;
    width: 80%;
    line-height: 22px;
    margin: 4px 0px 0px;
`;
