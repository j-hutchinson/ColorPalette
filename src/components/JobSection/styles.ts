import styled from 'styled-components'
import { colours } from '../../css/tokens'

export const StyledContainer = styled.section`
    border-radius: 20px;
    box-shadow: 20px 20px 60px rgb(0 0 0 / 50%), -20px -20px 60px rgb(255 255 255 / 5%);
    display: flex;
    justify-content: space-between;
    gap: 32px;
    padding: 64px 48px;
    max-width: 980px;
    margin: 80px auto;

    :first-child {
        margin-top: -60px;
    }

    @media screen and (max-width: 980px) {
        padding: 40px 20px;
        width: 560px;
    }

    @media screen and (max-width: 800px) {
         flex-direction: column-reverse;
    }
   
    @media screen and (max-width: 600px) {
        padding: 40px 20px;
        width: 335px;
    }

    @media screen and (max-width: 350px) {
        :first-child {
            margin-top: 0px;
        }
    }

  
`;

export const StyledCompanyName = styled.h3`
    margin: 0px;
    font-size: 22px;
    display: inline;

    @media screen and (max-width: 600px) {
        font-size: 18px;
    }
`;

export const StyledTitle = styled.h4`
    margin: 8px 0px 16px;
    color: ${colours.jobTitle};
    font-size: 18px;
    font-style: normal;

    @media screen and (max-width: 600px) {
        font-size: 16px;
    }
`;

export const StyledDate = styled.p`
    display: block;
    color: #9C9C9C;
    font-size: 16px;
    font-weight: 600;
    margin: 8px 0px 0px;
    text-overflow: ellipsis;
    word-wrap: break-word;
    overflow: hidden;
    max-height: 6.5em;

    @media screen and (max-width: 600px) {
        font-size: 14px;
        margin: 0px;
    }
`;

export const StyledDescription = styled.p`
    display: block;
    font-size: 16px;
    width: 90%;
    line-height: 22px;
    margin: 4px 0px 0px;
    
    @media screen and (max-width: 600px) {
        font-size: 14px;
    }
`;

export const StyledLogo = styled.div`
    display: flex;

    @media screen and (max-width: 800px) {
        margin: auto auto 16px;
    }
`;

export const StyledLink = styled.a`
    text-decoration: none;
`;
