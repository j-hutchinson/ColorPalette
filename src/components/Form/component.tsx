import { StyledForm, StyledIcon, StyledInput, StyledLabel } from './styles';
import { faRandom } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

interface Props {
  onRandom: () => void;
  onSubmit: (ev) => void;
}

const HueForm = ({ onRandom, onSubmit }: Props): JSX.Element => (
  <StyledForm onSubmit={onSubmit}>
    <StyledLabel>
      <StyledInput placeholder="Enter hue value" type="text" name="hueInput" />
    </StyledLabel>
    <input type="submit" value="Submit" />
    <StyledIcon onClick={onRandom}>
      <FontAwesomeIcon icon={faRandom} size="2x" />
    </StyledIcon>
  </StyledForm >
);

export default HueForm;
