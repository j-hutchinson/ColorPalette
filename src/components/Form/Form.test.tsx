import { shallow } from 'enzyme';
import Form from './component';
import { StyledForm, StyledIcon } from './styles';

describe('Form component', () => {
    const onRandom = jest.fn();
    const onSubmit = jest.fn();
    const event = {
        target: {
            value: 99,
        }
    }

    afterEach(() => {
        jest.resetAllMocks();
    })

    test('component matches snapshot', () => {
        expect.assertions(1);

        const wrapper = shallow(<Form onRandom={onRandom} onSubmit={onSubmit} />);

        expect(wrapper).toMatchSnapshot();
    });

    test('onSubmit called on form submission', () => {
        expect.assertions(2);

        const wrapper = shallow(<Form onRandom={onRandom} onSubmit={onSubmit} />);
        wrapper.find(StyledForm).simulate('submit', event)

        expect(onSubmit).toHaveBeenCalledTimes(1);
        expect(onSubmit).toHaveBeenCalledWith(event);
    });

    test('onRandom called on Icon click', () => {
        expect.assertions(1);

        const wrapper = shallow(<Form onRandom={onRandom} onSubmit={onSubmit} />);
        wrapper.find(StyledIcon).simulate('click');

        expect(onRandom).toHaveBeenCalledTimes(1);
    });
});