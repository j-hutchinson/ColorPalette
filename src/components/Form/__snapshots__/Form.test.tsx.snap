// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HueForm component component matches snapshot 1`] = `
<styled.form
  onSubmit={[MockFunction]}
>
  <styled.p>
    <styled.input
      name="hueInput"
      placeholder="Enter hue value"
      type="text"
    />
  </styled.p>
  <input
    type="submit"
    value="Submit"
  />
  <styled.span
    onClick={[MockFunction]}
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            512,
            512,
            Array [],
            "f074",
            "M504.971 359.029c9.373 9.373 9.373 24.569 0 33.941l-80 79.984c-15.01 15.01-40.971 4.49-40.971-16.971V416h-58.785a12.004 12.004 0 0 1-8.773-3.812l-70.556-75.596 53.333-57.143L352 336h32v-39.981c0-21.438 25.943-31.998 40.971-16.971l80 79.981zM12 176h84l52.781 56.551 53.333-57.143-70.556-75.596A11.999 11.999 0 0 0 122.785 96H12c-6.627 0-12 5.373-12 12v56c0 6.627 5.373 12 12 12zm372 0v39.984c0 21.46 25.961 31.98 40.971 16.971l80-79.984c9.373-9.373 9.373-24.569 0-33.941l-80-79.981C409.943 24.021 384 34.582 384 56.019V96h-58.785a12.004 12.004 0 0 0-8.773 3.812L96 336H12c-6.627 0-12 5.373-12 12v56c0 6.627 5.373 12 12 12h110.785c3.326 0 6.503-1.381 8.773-3.812L352 176h32z",
          ],
          "iconName": "random",
          "prefix": "fas",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size="2x"
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
  </styled.span>
</styled.form>
`;
